webapp/channels/src/components/header_footer_template_route.tsx

```diff
@@ -1,13 +1,17 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import React from 'react';
-import {Route} from 'react-router-dom';
+import React, {ComponentType} from 'react';
+import {Route, RouteProps, RouteComponentProps} from 'react-router-dom';
 
 const HeaderFooterTemplate = React.lazy(() => import('components/header_footer_template'));
 const LoggedIn = React.lazy(() => import('components/logged_in'));
 
-export const HFTRoute = ({component: Component, ...rest}) => (
+interface Props extends RouteProps {
+    component: ComponentType<RouteComponentProps>;
```

Comment:
Thanks for fighting with this file! Despite being so short, we're a lot of the code around this has complicated types, and we're definitely doing a bunch of things here which weren't meant to work with TS.

Unfortunately, while this version looks like it's probably correct, there's a couple places where the type checker is complaining about how the `PasswordResetSendLink` and `CreateTeam` components are being used with `HFTRoute` now, and I can't figure out a better way to fix them other than to change this to `ComponentType<any>`.

@M-ZubairAhmed Would that be okay with you?

Reply:
yes fine by me @notlelouch can you please do that change as above requested